### PR TITLE
add revert, closes #173

### LINKED ISSUE #173: Add DS#revert(resourceName, id)

### PR DESCRIPTION
**Pull request by internalfx**

### CODE REVIEW

File: src/datastore/sync_methods/index.js [hunk 1]
```diff
@@ -421,4 +421,27 @@ export default {
     // return resource from cache
     return resource.previousAttributes[id] ? DSUtils.copy(resource.previousAttributes[id]) : undefined;
   }
```
Comment (jmdobry) [inline src/datastore/sync_methods/index.js]: needs a comma after this curly brace

File: src/datastore/sync_methods/index.js [hunk 2]
```diff
@@ -421,4 +421,27 @@ export default {
...
+  /**
+   * Revert all attributes of the item with the given primary key th their previous values.
```
Comment (jmdobry) [inline src/datastore/sync_methods/index.js]: `key th their previous` should be `key to their previous`

File: src/datastore/sync_methods/index.js [hunk 3]
```diff
@@ -421,4 +421,27 @@ export default {
...
+    id = DSUtils.resolveId(definition, id);
+    if (!definition) {
+      throw new NER(resourceName);
+    } else if (!DSUtils._sn(id)) {
+      throw DSUtils._snErr('id');
+    }
+
+    definition.logFn('revert', id);
+
+    return definition.inject(this.previous(resourceName, id))
```
Comment (jmdobry) [inline src/datastore/sync_methods/index.js]: `this.previous` should be `_this.previous`

Comment (jmdobry) [inline src/datastore/sync_methods/index.js]: missing semi-colon at the end of the line

File: src/datastore/sync_methods/index.js [hunk 4]
```diff
@@ -420,5 +420,28 @@ export default {
 
     // return resource from cache
     return resource.previousAttributes[id] ? DSUtils.copy(resource.previousAttributes[id]) : undefined;
+  },
+  /**
+   * Revert all attributes of the item with the given primary key th their previous values.
```
Comment (jmdobry) [inline src/datastore/sync_methods/index.js]: `primary key th their previous values` should be `primary key to their previous values`

File: src/datastore/sync_methods/index.js [hunk 5]
```diff
@@ -420,5 +420,28 @@ export default {
...
+   *
+   * @param resourceName The name of the type of resource of the item.
+   * @param id The primary key of the item.
+   * @returns The reverted item
+   */
+  revert(resourceName, id) {
+    let _this = this;
+    let definition = _this.defs[resourceName];
+    let resource = _this.s[resourceName];
```
Comment (jmdobry) [inline src/datastore/sync_methods/index.js]: remove this line

**Comment by jmdobry** [comment]
Can you squash this PR into a single commit?

**Comment by internalfx** [comment]
Will do as soon as I finish the test

**Comment by internalfx** [comment]
bam

**Pull request merged**